### PR TITLE
Update ansible-pull To Use Python Functions

### LINKED ISSUE #26867: Have ansible-pull use python imports/functions instead of run_cmd
##### ISSUE TYPE
 - Feature Idea

##### COMPONENT NAME
pull

##### ANSIBLE VERSION
devel

##### SUMMARY
`run_cmd` is used in `lib/ansible/cli/pull.py` to call `ansible` and `ansible-playbook`. It would be less error prone and more performant to use python to import/call the corresponding modules instead of running shell commands.

### PR DESCRIPTION
**Pull request by akshay196**
##### SUMMARY
Using Python imports/functions to call ansible and ansible-playbook would be
less error prone and more performance instead of running shell
commands(using run_cmd).

Fixes #26867 

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ansible-pull

##### ANSIBLE VERSION
```
ansible 2.7.0.dev0
```
##### ADDITIONAL INFORMATION
* Use of AdHocCLI to run ansible command
(Thanks to @bcoca for this)

* Use of PlaybookCLI to run ansible-playbook

Signed-off-by: Akshay Gaikwad <akgaikwad001@gmail.com>

### CODE REVIEW

**Comment by akshaygin** [comment]
@bcoca @robinro I don't understand why Shippable CI is failing here. Could you just look into it, to figure out the problem? Let me know if I am missing something

**Comment by robinro** [comment]
@akshay196 The relevant failure can be seen e.g. in https://app.shippable.com/github/ansible/ansible/runs/75410/59/console:
```
2018-07-22 10:15:00 
2018-07-22 10:15:00 TASK [check for correct inventory] *********************************************
2018-07-22 10:15:00 task path: /tmp/ansible-testing-qIyBTmER3P/pull/local.yml:6
2018-07-22 10:15:00 fatal: [localhost]: FAILED! => {
2018-07-22 10:15:00     "msg": "testing for inventory content"
2018-07-22 10:15:00 }
2018-07-22 10:15:00 	to retry, use: --limit @/tmp/ansible-testing-qIyBTmER3P/pull/local.retry
2018-07-22 10:15:00 
2018-07-22 10:15:00 PLAY RECAP *********************************************************************
2018-07-22 10:15:00 localhost                  : ok=2    changed=0    unreachable=0    failed=1   
2018-07-22 10:15:00 
```

Looks like your change broke how ansible-pull is expected to deal with inventories.

**Comment by akshaygin** [comment]
@robinro Thank you for help. 
I think, failure because of inventory file is solved now.
```
2018-07-23 17:58:37 TASK [check for correct inventory] *********************************************
2018-07-23 17:58:37 ok: [localhost] => {
2018-07-23 17:58:37     "msg": "testing for inventory content"
2018-07-23 17:58:37 }
2018-07-23 17:58:37 
```
Looking into other reasons of CI failure.

**Comment by akshaygin** [comment]
@robinro Could you review this PR now? let me know if there are changes need to be done.

File: lib/ansible/cli/pull.py [hunk 1]
```diff
@@ -116,8 +112,7 @@ def parse(self):
         self.parser.add_option('--accept-host-key', default=False, dest='accept_host_key', action='store_true',
                                help='adds the hostkey for the repo url if not already added')
         self.parser.add_option('-m', '--module-name', dest='module_name', default=self.DEFAULT_REPO_TYPE,
-                               help='Repository module name, which ansible will use to check out the repo. Choices are %s. Default is %s.'
-                                    % (self.REPO_CHOICES, self.DEFAULT_REPO_TYPE))
```
Comment (mscherer) [inline lib/ansible/cli/pull.py]: Why is the error message changed here ?

Comment (akshay196) [reply lib/ansible/cli/pull.py]: My mistake. I think I unknowingly changed that. I will update PR to keep the previous message. @mscherer Thank you.

**Comment by jimi-c** [comment]
Hi @akshay196, @bcoca had created a more updated PR which also addresses this here https://github.com/ansible/ansible/pull/31555, though it also needs to be rebased. We'll go ahead and close this, and if you're still interested in working on this I'd recommend checking out that PR.

If you have any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:

   * https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
   * https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code
   * IRC: #ansible or #ansible-devel on irc.freenode.net

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular issue is resolved.

Thank you!

**Pull request closed without merging**